core/datapool/datapool-core/src/main/kotlin/io/holunda/polyflow/datapool/core/business/DataEntryAggregate.kt

```diff
@@ -118,4 +133,21 @@ class DataEntryAggregate() {
     // Don't use AggregateLifecycle.markDeleted() because then the combination of entryType / entryId is then really deleted forever
     this.deleted = true
   }
+
+  /**
+   * React on anonymized event.
+   */
+  @EventSourcingHandler
+  fun on(event: DataEntryAnonymizedEvent) {
+    if (this.deleted) {
+      this.deleted = false
```

Comment:
Is it possible? If the aggregate is already marked as deleted, the command handler would throw an exception, so the event is never emitted. If it is not, this check makes no sense. Or am I missing something?

Reply:
To be honest I do not completely understand the aggregation. I kind of followed what happens with `UpdateDataEntryCommand`
```
/**
   * Handle update.
   */
  @CommandHandler
  fun handle(command: UpdateDataEntryCommand, @Autowired deletionStrategy: DeletionStrategy) {
    if (deletionStrategy.strictMode()) {
      if (deleted) {
        throw AggregateDeletedException(this.dataIdentity, "The data entry has already been deleted")
      }
    }
    AggregateLifecycle.apply(
      command.updatedEvent()
    )
  }
```
and then
```
  /**
   * React on updated event.
   */
  @EventSourcingHandler
  fun on(event: DataEntryUpdatedEvent) {
    if (this.deleted) {
      this.deleted = false
    }
    if (logger.isDebugEnabled) {
      logger.debug { "Updated $dataIdentity." }
    }
    if (logger.isTraceEnabled) {
      logger.trace { "Updated $dataIdentity with: $event" }
    }
  }
```

Reply:
probably ok to just remove the
```
    if (this.deleted) {
      this.deleted = false
    }
```
because otherwise with lax deletion policy the data entry would be "undeleted". Probably expected behavior would be to anonymize it but keep it deleted if it was already deleted

Reply:
removed the EventSourceHandler completely